sys/fork.c

```diff
@@ -41,24 +42,24 @@ int do_fork(void) {
   newtd->td_prio = td->td_prio;
 
   /* Now, prepare a new process. */
-  assert(td->td_proc);
-  proc_t *proc = proc_create(newtd, td->td_proc);
+  proc_t *child = proc_create(newtd, parent);
```

Comment:
For the record:
proc_create registers process in some global lists and then we change it's fields.
But it will be fixed with #525

Reply:
ok